### ISSUE TITLE
Reactivity?

### ISSUE THREAD

**Issue by VladShcherbin** [jagi/meteor-astronomy#2]
Nice package. The question is - when I get the `post` instance - will it reactively update when its data change or do I need to call `reload` method ?

**Comment by ccorcos** [comment]
good question. I don't quite understand the point of `reload`. Or does it just reset the post by removing all modifications?

**Comment by jagi** [comment]
Thanks for pointing it out. I will try to make it clear in the documentation. I will copy here my answer from [Crater.io](https://crater.io/posts/MT9dPLYiydEWmqYi5/comment/tCQH3LGahRQDhCkNs).

"Good question. You don't have to do anything more than you would do normally to display documents from collection to make it reactive. Cursor returned from collection is standard Mongo cursor and it's reactive. However single objects are not reactive the same like standard JS object that would be returned from Mongo collection. If you want to have reactivity you always have to use cursors. However I'm thinking about implementing reactivity on the level of object, so that any change to document would have immediate effect in the UI, even if document is not saved in the collection. However it would be an option and not standard behaviour.

Is it the feature you would like to see? If yes, please could you give some example in what cases is it necessary to have such feature?"

And to answer your question about `reload` function. It just refreshes document with its new values loaded from database. Single object is not reactive (yet) so it has no idea that something changed in the collection. And there are some situations where we want to make sure that we work with the newest state of the object.

**Comment by VladShcherbin** [comment]
@jagi thanks for the answer. With the examples, it is clear now, how the package is used. Since we are getting cursors and not the model instances, the reactivity is present.

**Comment by jagi** [comment]
Yes exactly :). If you have any other question feel free to ask.

**Comment by jagi** [comment]
If you can I would like you also to tell your opinion on this issue https://github.com/jagi/meteor-astronomy/issues/3

Thanks :)

**Comment by darko-mijic** [comment]
@jagi, is there a plan to implement object level reactivity as You have mentioned in April:

> However I'm thinking about implementing reactivity on the level of object, so that any change to document would have immediate effect in the UI, even if document is not saved in the collection. However it would be an option and not standard behaviour.

I am currently considering the best solution for object level reactivity with current state of Astronomy (1.0) so I was wondering do You have any new thoughts on this subject...

Thanks :)

**Comment by jagi** [comment]
It's not implemented in 1.0, however have it in plans for Astronomy 2.0.

I'm preparing for it by creating packages like [ReactiveArray](https://atmospherejs.com/jagi/reactive-array). I already have [ReactiveMap](https://atmospherejs.com/jagi/reactive-map). I want it to be built in, closely integrated with a data and easy to use.

Right now I'm going to work on `forms` and `relations` packages which are "must have". After that I will be ready to focus on improving Astronomy.

If you want reactivity right now, you can create something that will work (I've tested it and it works). In simplified form it could look like in the example below. However it only works with top level fields. Nested fields won't be updated reactively. That's why I want it to be implemented in Astronomy 2.0. Maybe I will create module for a reactivity for Astronomy 1.0.

```js
Item = Astro.Class({
  name: 'Item',
  collection: Items,
  fields: {
    /* ... */
  },
  methods: {
    getReactiveContext: function() {
      var self = this;
      var Class = self.constructor;
      var fieldsNames = Class.getFieldsNames();
      _.each(fieldsNames, function(fieldName) {
        self._reactiveDoc.set(self[fieldName]);
      });
      return self._reactiveDoc;
    }
  },
  events: {
    beforeInit: function() {
      this._reactiveDoc = this._reactiveDoc || new ReactiveMap();
    },
    afterChange: function(e) {
      var fieldName = e.data.fieldName;
      this._reactiveDoc.set(fieldName, this[fieldName]);
    }
  }
});
```

 You have to add `jagi:reactive-map` package to your project.
